src/System.Runtime.Extensions/tests/Performance/Perf.Path.cs

```diff
@@ -62,5 +62,169 @@ public void GetDirectoryName()
+                    {
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                        Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension); Path.ChangeExtension(testPath, extension);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetExtension(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                        Path.GetExtension(testPath); Path.GetExtension(testPath); Path.GetExtension(testPath);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetFileNameWithoutExtension(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                        Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath); Path.GetFileNameWithoutExtension(testPath);
+                    }
+        }
+
+        [Benchmark]
+        [InlineData(10000)]
+        [InlineData(20000)]
+        [InlineData(30000)]
+        public void GetFullPath(int innerIterations)
+        {
+            PerfUtils utils = new PerfUtils();
+            string testPath = utils.GetTestFilePath();
+            foreach (var iteration in Benchmark.Iterations)
+                using (iteration.StartMeasurement())
+                    for (int i = 0; i < innerIterations; i++)
+                    {
+                        Path.GetFullPath(testPath); Path.GetFullPath(testPath); Path.GetFullPath(testPath);
```

Comment:
> That could skew the results.

Yep, for all of these. They are very much bound to the length of the path- of which GetFullPath ends up playing a part in most of the APIs as most call GFP.